### DISCUSSION TITLE
Convert the cv page into PDF for download

### DISCUSSION THREAD

**Discussion by ghltshubh** [alshedivat/al-folio#1876] [Ideas]
It would be really convenient if the cv page could be directly converted into a downloadable pdf. It would get rid of maintaining cv in two formats.

**Comment by george-gca** [comment]
If you create your cv information as a [json file](https://github.com/alshedivat/al-folio/blob/master/assets/json/resume.json), you could easily update both the site and the pdf file. Maybe you could even use a GitHub Action similar to [this one](https://github.com/marketplace/actions/jsonresume-export) to automatically recreate the pdf version. 

Another alternative is for you to use a [GitHub Action](https://github.com/marketplace/actions/html-to-pdf) to convert the html file straight to pdf.

**Reply by ghltshubh** [reply]
Thanks for the solution but IMHO maintaining a json resume especially for a researcher is too much work. The resume only grows with time and every opening & closing `"` and `{` in the json could become intractable which is prone to error even with help of formatters. Though I think writing it in a markdown or maybe yaml would be much more convenient.

**Reply by george-gca** [reply]
I see your point. Then why not give [this](https://github.com/marketplace/actions/html-to-pdf) a try?

**Reply by ghltshubh** [reply]
Sure! Will give it a try. 👍

**Reply by ghltshubh** [reply]
I was trying to use html to pdf github action but I am not sure what's the location of the /cv html page because it is generated at the run time from cv.md I am assuming.

**Reply by george-gca** [reply]
Actually not at runtime, but during [build](https://github.com/alshedivat/al-folio/blob/85f2590a2119ec57063845d628221155044ebea8/.github/workflows/deploy.yml#L36). Jekyll is a [static site generator](https://en.wikipedia.org/wiki/Static_site_generator), meaning it can't create content at runtime, just during build ("compile") time.

After build a directory `_site/` is created with the actual pages. The result is then pushed by a later [GitHub Action](https://github.com/alshedivat/al-folio/blob/85f2590a2119ec57063845d628221155044ebea8/.github/workflows/deploy.yml#L44) to the `gh-pages` branch (which is [the default](https://github.com/JamesIves/github-pages-deploy-action#optional-choices) for the action). The result can be seen [here](https://github.com/alshedivat/al-folio/blob/gh-pages/cv/index.html).

So you will either need to fetch the page from the `_site/` directory or from the `gh-pages` branch.

**Reply by ghltshubh** [reply]
Actually, I am not building the site locally. I am using jekyll github actions to build and deploy the site and I am very new to github actions so don't exactly know how they work.

**Reply by george-gca** [reply]
You will probably need to fetch the page from the `gh-pages` branch, in `cv/index.html`.

**Reply by ghltshubh** [reply]
I only have the main branch. It doesn't generate any other branches.

**Reply by george-gca** [reply]
Is your site being correctly deployed online?

**Reply by ghltshubh** [reply]
Yes it's here ghltshubh.github.io

**Reply by george-gca** [reply]
Just checked your repo, and your Github action to deploy the page is different from our solution. You need to check the action that copy the built content from jekyll output to the Github page to find where is the built cv page.

**Reply by ghltshubh** [reply]
I am looking for it as well but couldn't find it. Kind of still getting used to github actions. I am just using pdf as of now. 😅
